### PR TITLE
Update to latest clippy release and address all clippy errors and warnings

### PR DESCRIPTION
**Pull request by mrohera**
Changes include updating the scripts to use the latest clippy version and also ensuring that clippy will be run locally after it is installed via rustup as opposed to running within a docker image.

The other changes are fixes to warnings and errors reported by clippy.

### CODE REVIEW

File: edgelet/build/linux/clippy.sh [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
 
 ###############################################################################
-# This script runs cargo clippy on your project
+# This script runs cargo clippy on your project. This script assumes that the
+# nigtly toolchain is installed.
```
Comment (avranju) [inline edgelet/build/linux/clippy.sh]: Nit: Typo: _nightly_.

File: edgelet/win-logger/src/lib.rs [hunk 2]
```diff
@@ -60,8 +60,11 @@ impl EventLogger {
     }
 
     pub fn init(self) -> Result<(), Error> {
+        // log::set_boxed_logger(Box::new(self))
```
Comment (myagley) [inline edgelet/win-logger/src/lib.rs]: Should we remove this commented code?

Comment (mrohera) [reply edgelet/win-logger/src/lib.rs]: Sure will remove.

Comment (mrohera) [reply edgelet/win-logger/src/lib.rs]: Fixed.

**Comment by avranju** [comment]
Can you squash all the commits please? Otherwise the auto squash goes and puts the commit on each individual commit as additional detail on the commit message. Which may not be something that we want.

I can approve now but that will get discarded when you do the squash commit.

**Pull request merged**